### PR TITLE
feat: added configuration capabilities for sharedMailbox and subject

### PR DESCRIPTION
**Pull request by bearpersson**
## Explain the changes you’ve made

Added configuration capabilities for sharedMailbox and subject.

## Explain why these changes are made

To avoid hard-coded values and to be able to use the application for diffrent purposes in the organization.

## Explain your solution

Added sharedMailbox and subject as data attributes to the root element as we do with the data endpoint.

## Tested environments

- [] Your local environment.
- [] Staging environment.

## Screenshots

(Optional)

Screenshots are helpful for UI-related changes. It could be an before and after change as an example.
Even backend code can benefit from a screenshot of the net change.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

File: source/php/AcfFields/json/mhbooking-module.json [hunk 1]
```diff
@@ -1,47 +1,80 @@
...
-    "location": [
-        [
             {
-                "param": "post_type",
-                "operator": "==",
-                "value": "mod-mhbooking"
-            }
-        ],
-        [
+                "key": "REPLACE1",
```
Comment (bearpersson) [inline source/php/AcfFields/json/mhbooking-module.json:23]: How do we genereate the key here?

- [ ] Fix

File: public/index.html [hunk 2]
```diff
@@ -24,26 +21,22 @@
...
-                    <h2
-                      class="c-typography c-card__heading c-typography__variant--p"
-                    >
-                      Boka ditt möte
-                    </h2>
-                    <div data-endpoint="xxxxx" id="mh-mount-booking"></div>
+                    <h2 class="c-typography c-card__heading c-typography__variant--p">Boka ditt möte</h2>
+                    <div
+                      data-endpoint="xxxxx"
+                      data-sharedmailbox="datatorget_testgrupp@helsingborgdemo.onmicrosoft.com"
```
Comment (karlknowit) [inline public/index.html:36]: Not sure if data-sharedmailbox and data-subject should be hard-coded here?

File: source/js/services/BookingService.ts [hunk 3]
```diff
@@ -151,7 +153,7 @@ const buildBookingRequest = (timeSlot: TimeSlot, formData: FormData): BookingReq
     date: timeSlot.date,
     endTime: `${timeSlot.date}T${timeSlot.endTime}`,
     startTime: `${timeSlot.date}T${timeSlot.startTime}`,
-    subject: 'Volontärsamtal',
```
Comment (karlknowit) [inline source/js/services/BookingService.ts:154]: Prefer property shorthand

```js
 return {
   ...
   subject,
}
```

File: source/php/AcfFields/json/mhbooking-module.json (hunk 1 above)
Comment (karlknowit) [reply source/php/AcfFields/json/mhbooking-module.json:23]: Do we have an answer for this?

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/php/AcfFields/json/mhbooking-module.json (hunk 1 above)
Comment (bearpersson) [reply source/php/AcfFields/json/mhbooking-module.json:23]: I interpreted it as any sting is ok. I change it to something more relevant.

File: public/index.html (hunk 2 above)
Comment (bearpersson) [reply public/index.html:36]: Do you meen that should hide it as we do with `data-endpoint` and configure it locally? This file is not a part of the application and the `div` is just a mountingpoint for the app.

File: source/js/services/BookingService.ts (hunk 3 above)
Comment (bearpersson) [reply source/js/services/BookingService.ts:154]: Fixed

File: public/index.html (hunk 2 above)
Comment (karlknowit) [reply public/index.html:36]: Yes, thats exactly what I meant.

Comment (bearpersson) [reply public/index.html]: Okey fixed :)

**Review by karlknowit — APPROVED** [review]

**Pull request merged**